File: app/src/main/java/com/sanislo/movieapp/di/upcoming/UpcomingMoviesFragmentProvider.java
package com.sanislo.movieapp.di.upcoming;

import com.sanislo.movieapp.presentation.upcomingMovies.UpcomingMoviesFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class UpcomingMoviesFragmentProvider {

    @ContributesAndroidInjector(modules = UpcomingMoviesFragmentModule.class)
    abstract UpcomingMoviesFragment provideUpcomingMoviesFragment();
}

File: app/src/main/java/com/sanislo/movieapp/di/upcoming/UpcomingMoviesFragmentProvider.kt
package com.sanislo.movieapp.di.upcoming

import com.sanislo.movieapp.presentation.upcomingMovies.UpcomingMoviesFragment

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class UpcomingMoviesFragmentProvider {

    @ContributesAndroidInjector(modules = [UpcomingMoviesFragmentModule::class])
    internal abstract fun provideUpcomingMoviesFragment(): UpcomingMoviesFragment
}


